File: golang/21. 合并两个有序链表.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    head:=&ListNode{-1,nil}
    p:=head
    p1, p2:=list1, list2
    for p1!=nil && p2!=nil {
        if p1.Val < p2.Val {
            p.Next=p1
            p1=p1.Next
        } else {
            p.Next=p2
            p2=p2.Next
        }
        p=p.Next
    }
    if p1!=nil {
        p.Next=p1
    }
    if p2!=nil {
        p.Next=p2
    }
    return head.Next
}

File: 刷题/cpp/21. 合并两个有序链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* p=list1;
        ListNode* q=list2;
        ListNode* list3=new ListNode();
        ListNode* cur=list3;
        while(p!=nullptr && q!=nullptr){
            if(p->val<q->val){
                ListNode* t=new ListNode(p->val);
                cur->next=t;
                cur=t;
                p=p->next;
            }else{
                ListNode* t=new ListNode(q->val);
                cur->next=t;
                cur=t;
                q=q->next;
            }
        }
        while(p!=nullptr){
            ListNode* t=new ListNode(p->val);
            cur->next=t;
            cur=t;
            p=p->next;
        }
        while(q!=nullptr){
            ListNode* t=new ListNode(q->val);
            cur->next=t;
            cur=t;
            q=q->next;
        }
        return list3->next;
    }
};